### PR TITLE
[TPID CONFIG]TPID attribute Yang model and default TPID for Minigraph to configDB Changes

### PR DESCRIPTION
**Pull request by gechiang**
#### Why I did it
This PR is part of the TPID CONFIG SONiC feature set to support port/LAG TPID configuration.  
TPID HLD PR: (https://github.com/Azure/SONiC/pull/681)

There are total of 3 PRs across different submodules to achieve this feature support.

- SONiC-Utilities (https://github.com/Azure/sonic-utilities/pull/1618) - To handle TPID CLI related changes.
- SONiC-Swss (https://github.com/Azure/sonic-swss/pull/1747) - To handle Orchagent changes triggered from TPID configuration changes to interact with SYNCd SAI.
- and this PR which handles the new port/LAG TPID attribute on Yang model and Minigraph to configDB changes.

Please refer to the TPID HLD listed above for more details on the TPID configuration feature support.
Here is a quick summary for this PR changes:
By default TPID for port/LAG is 0x8100.  With HW SKU that has capability to handle TPID configuration user is allowed to configure the port/LAG to non default TPID values such as (0x9100, 0x9200, 0x88A8). 
This PR is to define this new TPID attribute for Port and LAG in the Yang model as well as to ensure all the ports/LAGs in the config DB generated from Minigraph should have a default TPID value of 0x8100 .

#### How to verify it
Test code were added for the Yang model to test the new TPID attribute in PORT and Portchannel.
Test code were also updated in test config gen, minigraph as well as multi-asic config gen to ensure the expected default TPID is in place.
Complete feature testing of TPID functionality were also performed on DUTs that has SAI support for TPID attribute to ensure the feature is working properly.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog
TPID port/LAG attribute Yang model and default TPID for Minigraph to configDB Changes

### CODE REVIEW

File: src/sonic-yang-models/tests/files/sample_config_db.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/files/sample_config_db.json:369]: We do not need to add it in all ports, just add it in 1 or 2 ports.

File: src/sonic-yang-models/yang-models/sonic-portchannel.yang
Comment (praveen-li) [inline src/sonic-yang-models/yang-models/sonic-portchannel.yang:115]: See, if we can move type to stypes.yang, and use that typedef here.

File: src/sonic-yang-models/yang-models/sonic-port.yang
Comment (gechiang) [inline src/sonic-yang-models/yang-models/sonic-port.yang:100]: 1. Add description field to describe what this attribute is about
2. move to common so both port and LAG can share to avoid duplication

Comment (gechiang) [reply src/sonic-yang-models/yang-models/sonic-port.yang:100]: we will not follow the open config model for this... for simplicity handling in config DB.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/portchannel.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/portchannel.json:13]: We can avoid updating old tests, since it is not mandatory field. Just add it in new tests.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json:93]: We do not need extra test, once type is moved in stypes.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json
Comment (praveen-li) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json:15]: No need to update old tests.

File: src/sonic-yang-models/yang-models/sonic-port.yang
Comment (gechiang) [reply src/sonic-yang-models/yang-models/sonic-port.yang:100]: In the future when stateDB validation method is agreed, we can come back to this and enhance it to check capability as part of the yang model validation for TPID attribute

File: src/sonic-config-engine/tests/test_multinpu_cfggen.py
Comment (dgsudharsan) [inline src/sonic-config-engine/tests/test_multinpu_cfggen.py:152]: Can you please correct the alignment in this file?

File: src/sonic-yang-models/tests/files/sample_config_db.json
Comment (gechiang) [reply src/sonic-yang-models/tests/files/sample_config_db.json:369]: Going forward after this PR is approved and merged, the config DB when generated from the Minigraph will always have TPID 0x8100.  It is similar to the MTU handling where there is a default value set in config DB.
I like to keep this change to set TPID attribute for all the interfaces (port and LAG). this way it is more in sync when user inspect config DB.

File: src/sonic-yang-models/yang-models/sonic-portchannel.yang
Comment (gechiang) [reply src/sonic-yang-models/yang-models/sonic-portchannel.yang:115]: Agreed. Will have this changed in next commit.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json
Comment (gechiang) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json:93]: Not sure I understood your comment.  moving the tpid type to a common typedef allows sharing of the same definition  but testing is still required isn't it?  LThis was added so that we can see that the Yang model correctly catches any attempt of using a wrong TPID value that is not specified in the model.  Please clarify ...
Thanks!

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/portchannel.json
Comment (gechiang) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/portchannel.json:13]: I am actually leveraging this same test config file for the new attribute TPID testing.  Are you saying that I should generate a new test file just for TPID purpose?  since TPID is a new attribute for both port and portchannel, I thought of just expanding on existing test case  instead of generating a new file for just one TPID test...
There are no other tests needed other than ensure that the model can catch invalid TPID values.

File: src/sonic-config-engine/tests/test_multinpu_cfggen.py
Comment (gechiang) [reply src/sonic-config-engine/tests/test_multinpu_cfggen.py]: fixed.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json
Comment (gechiang) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/port.json:15]: TPID will be present in all config DB after thei feature is checked in so making sure it is consistently present in all interface configs.

**Review by lguohan — APPROVED** [review]

**Pull request merged**
